### PR TITLE
Grammar, spelling, and punctuation cleanup.

### PR DESCRIPTION
**Pull request by ShawnMilo**
Additonal cleanup inspired by your great effort. 

Thanks!

### CODE REVIEW

**Comment by ShawnMilo** [comment]
Note: Many of the changes are spaces around `&mdash;` which were missing.

File: README.md [hunk 1]
```diff
@@ -414,14 +414,14 @@ func (rmqch *RMQChannel) QueueDeclare(opts QueueOptions) (Queue, error) {
...
 
-The problem with using global variables is that we make it very difficult for programmers to understand the current state of a variable. If a variable is global and mutable, then by definition, its value can be changed by any part of the codebase. At no point can you guarantee that this variable is going to be a specific value... And that's a headache for everyone. This is yet another example of a trivial problem that's exacerbated when the codebase expands.
+The problem with using global variables is that we make it very difficult for programmers to understand the current state of a variable. If a variable is global and mutable, then by definition, its value can be changed by any part of the codebase. At no point can you guarantee that this variable is going to contain a specific value; and that's a headache for everyone. This is yet another example of a trivial problem that's exacerbated when the codebase expands.
```
Comment (AleksandrHovhannisyan) [inline README.md]: I agree with this change :) The only thing I'd request is to replace the semicolon with a comma.

File: README.md [hunk 2]
```diff
@@ -414,14 +414,14 @@ func (rmqch *RMQChannel) QueueDeclare(opts QueueOptions) (Queue, error) {
 } 
 ```
 
-Basically, we create a new structure named `RMQChannel` that contains the `amqp.Channel` type, which has the `QueueDeclare` method. We then create our own version of this method, which essentially just calls the old version of the RabbitMQ library function. Our new method has all the advantages described before, and we achieved this without actually having to change any of the code in the RabbitMQ library.
+Basically, we create a new structure named `RMQChannel` that contains the `amqp.Channel` type, which has the `QueueDeclare` method. We then create our own version of this method, which essentially just calls the old version from the RabbitMQ library. Our new method has all the advantages described above, and we achieved this without having to change the RabbitMQ library.
```
Comment (AleksandrHovhannisyan) [inline README.md]: Ah, good catch :)

File: README.md [hunk 3]
```diff
@@ -615,18 +615,18 @@ func main() {
...
 
 >  NOTE: There is a fantastic explanation of how to handle the abstraction in client libraries, taken from the talk [AWS re:Invent 2017: Embracing Change without Breaking the World (DEV319)](https://www.youtube.com/watch?v=kJq81Y7OEx4).
 
 ## Clean Go
 
-This section will describe some less generic aspects of writing clean Go code, but rather be discussing aspects that are very go specific. Like the previous section, there will still be a mix of generic and specific concepts being discussed, however, this section marks the start of the document, where the document changes from a generic description of clean code with Go examples, to Go specific descriptions, based on clean code principles.
+This section will describe some less generic aspects of writing clean Go code, but rather be discussing aspects that are very Go-specific. Like the previous section, there will still be a mix of generic and specific concepts being discussed, however, this section marks the point in this document where the goal changes from a generic description of clean code with Go examples, to Go-specific descriptions based on clean code principles.
```
Comment (AleksandrHovhannisyan) [inline README.md]: Note: This section was actually rewritten. I think you may have been working with an outdated version of the branch :)

**Pull request closed without merging**